### DISCUSSION TITLE
Guidance on data files needed by ports

### DISCUSSION THREAD

**Discussion by jumpinjackie** [microsoft/vcpkg#20700] [Q&A]
I'd like to make the CS-Map coordinate system transformation library (https://trac.osgeo.org/csmap) available as a vcpkg port.

While the library itself is trivial to make available as a vcpkg port, my questions surround the data files this library needs to function.

Namely ... these files are **huge**, totaling around 1.5 GB (uncompressed). If CS-Map were made available as a vcpkg port, it needs these data files to function.

What would be the proper guidance around such libraries? Should we include the data files with the port? If we do, does vcpkg support the notion of a shared path for data files? Because I'd hate the thought of having 1.5GB of data files duplicated for each vcpkg triplet configuration of this library.

**Comment by mloskot** [comment]
Perhaps the auxiliary data could be available via an optional feature
```
cs-map -> cs-map[core]
cs-map[data]
```

where `vcpkg install cs-map` pulls the core only and `vcpkg install cs-map[data]` means user explicitly agreed to download the large archive.

**Comment by vicroms** [comment]
This is a pretty old discussion but it should be possible to avoid duplicating the data by using a host port. 

```
{
  "name": "csmap",
  "version": "1.0.0",
  "description": "the csmap library",
  "features": {
    "data": {
      "description": "installs csmap data (approximately 1.5GB)",
      "dependencies": [
        {
          "name": "csmap-data",
          "host": true
        }
      ]
    }
  }
}
```
